core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -881,17 +881,74 @@ object JsonAST {
      */
     val compactJs = RenderSettings(0, jsEscapeChars)
   }
+
+  /**
+   * Parent trait for double renderers, which decide how doubles contained in
+   * a JDouble are rendered to JSON string.
+   */
+  sealed trait DoubleRenderer extends Function1[Double,String] {
+    def apply(double: Double): String
+  }
+  /**
+   * A `DoubleRenderer` that renders special values `NaN`, `-Infinity`, and
+   * `Infinity` as-is using `toString`. This is not strictly valid JSON, but
+   * JavaScript can eval it. Other double values are also rendered the same
+   * way.
+   *
+   * Usage is not recommended.
+   */
+  case object RenderSpecialValuesAsIs extends DoubleRenderer {
+    def apply(double: Double): String = {
+      double.toString
+    }
+  }
+  /**
+   * A `DoubleRenderer` that renders special values `NaN`, `-Infinity`, and
+   * `Infinity` as `null`. Other doubles are rendered normally using
+   * `toString`.
+   */
+  case object RenderSpecialValuesAsNull extends DoubleRenderer {
+    def apply(double: Double): String = {
+      if (double.isNaN || double.isInfinity) {
+        "null"
+      } else {
+        double.toString
+      }
+    }
+  }
+  /**
+   * A `DoubleRenderer` that throws an `IllegalArgumentException` when the
+   * special values `NaN`, `-Infinity`, and `Infinity` are encountered. Other
+   * doubles are rendered normally using `toString`.
+   */
+  case object FailToRenderSpecialValues extends DoubleRenderer {
+    def apply(double: Double): String = {
+      if (double.isNaN || double.isInfinity) {
+        throw new IllegalArgumentException(s"Double value $double is not renderable to JSON.")
```

Comment:
Can we change this to `Double value $double is not renderable to JSON when FailToRenderSpecialDoubleValues is used for rendering`